mermin/src/span/ebpf_guard.rs

```diff
@@ -118,7 +118,8 @@ impl Drop for EbpfFlowGuard {
                     );
 
                     // Metrics: Flow expired via guard cleanup (error path)
-                    inc_flows_expired("unknown", "guard_cleanup");
+                    inc_producer_flow_spans("unknown", "guard_cleanup");
+                    dec_flows_active("unknown");
```

Comment:
Something in my gut tells me we shouldn't be doing that here.

When does the guard drop? On exit of the program or is there other conditions?